modules/postfix.lua

```diff
@@ -0,0 +1,522 @@
+          * l.Cg((l.P(1) - l.P' status=')^1, 'postfix_keyvalue_data')
+          * l.P' status='
+          * l.Cg(l.alnum^1, 'postfix_status')),
+  -- lmtp = smtp,
+  pickup = l.Ct(postfix_queueid_cg
+         * l.P': uid='
+         * l.Cg((l.P(1) - l.P' from=')^1, 'postfix_uid')
+         * l.P' from=<'
+         * l.Cg((l.P(1) - l.P'>')^1, 'postfix_from')
+         * (l.P'> orig_id='
+           * l.Cg(l.P(1)^1, 'postfix_orig_id'))^-1
+         ),
+  tlsproxy = l.Ct(postfix_tlsproxy_conn_cg),
+  master = l.Ct(postfix_master_start_cg
+         + postfix_master_exit_cg),
+  bounce = l.Ct(postfix_bounce_notification_cg),
+  sendmail = l.Ct(postfix_warning_cg),
+  postdrop = l.Ct(postfix_warning_cg),
+  scache = l.Ct(postfix_scache_lookups_cg
+         + postfix_scache_simultaneous_cg
+         + postfix_scache_timestamp_cg),
+  trivial_rewrite = l.Ct(postfix_warning_cg),
+  tlsmgr = l.Ct(postfix_warning_cg),
+  ['local'] = l.Ct(postfix_keyvalue_greedy_cg),
+}
+postfix_patterns['lmtp'] = postfix_patterns['smtp']
+
+--local name = l.C(l.alpha^1) * space
+--local value = l.C((l.P(1) - l.P',')^1)
+--local sep = l.S(",;") * space
+--local pair = l.Cg(name * "=" * space * value) * sep^-1
+--local queueid = lpeg.Cf(lpeg.Ct"" * l.Cg(lpeg.Cc'queueid' * l.C(l.xdigit ^ 1)) * l.P': ', rawset)
+--local postfix_grammar = l.Cf(queueid * pair^0, rawset)
+
+
+
+local postfix_programname = l.Ct(l.P'postfix'
+                           * (l.P(1) - l.P'/')^0
+                           * '/'
+                           * l.Cg(l.P(1)^1, 'daemon'))
+
+local integer_fields = {
+  "postfix_anvil_cache_size",
+  "postfix_anvil_conn_count",
+  "postfix_anvil_conn_rate",
+  "postfix_client_port",
+  "postfix_nrcpt",
+  "postfix_postscreen_cache_dropped",
+  "postfix_postscreen_cache_retained",
+  "postfix_postscreen_dnsbl_rank",
+  "postfix_relay_port",
+  "postfix_server_port",
+  "postfix_size",
+  "postfix_status_code",
+  "postfix_termination_signal",
+  "postfix_uid",
+}
+local float_fields = {
+  "postfix_delay",
+  --"postfix_delay_before_qmgr",
+  --"postfix_delay_conn_setup",
+  --"postfix_delay_in_qmgr",
+  --"postfix_delay_transmission",
+  --"postfix_postscreen_violation_time",
+}
+
+
+function M.postfix_match(programname, message, extract_keyvalue_data)
+  local programtable = postfix_programname:match(programname)
+  local ret = nil
+  if not programtable then
+    return nil
+  end
+  if postfix_patterns[programtable['daemon']] then
+    ret = postfix_patterns[programtable['daemon']]:match(message)
+  -- below are fake patterns used by tests
+  elseif programtable['daemon'] == 'relay_info' then
+    ret = l.Ct(postfix_relay_info_cg):match(message)
+  elseif programtable['daemon'] == 'delays' then
+    ret = postfix_delays_ct:match(message)
+  elseif programtable['daemon'] == 'keyvalue_data' then
+    ret = kv_grammar:match(message)
+  end
+  -- extract postfix_keyvalue_data
+  if extract_keyvalue_data and ret and ret['postfix_keyvalue_data'] then
```

Comment:
Since the keys don't have reserved characters you can use the shorthand.  `ret.postfix_keyvalue_data` etc...